<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.433">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>DNA Breathing in ChIP-seq and gcPBM Data - Analysis of gcPBM data</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script src="site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">DNA Breathing in ChIP-seq and gcPBM Data</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="./index.html" rel="" target="">
 <span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./about.html" rel="" target="">
 <span class="menu-text">About</span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools ms-auto">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#overview-on-the-dataset-and-data-processing" id="toc-overview-on-the-dataset-and-data-processing" class="nav-link active" data-scroll-target="#overview-on-the-dataset-and-data-processing">Overview on the Dataset and Data Processing</a>
  <ul class="collapse">
  <li><a href="#bubbles" id="toc-bubbles" class="nav-link" data-scroll-target="#bubbles">Bubbles</a></li>
  <li><a href="#motifs" id="toc-motifs" class="nav-link" data-scroll-target="#motifs">Motifs</a></li>
  <li><a href="#processed-data" id="toc-processed-data" class="nav-link" data-scroll-target="#processed-data">Processed Data</a></li>
  </ul></li>
  <li><a href="#machine-learning-investigation" id="toc-machine-learning-investigation" class="nav-link" data-scroll-target="#machine-learning-investigation">Machine Learning Investigation</a></li>
  <li><a href="#statistical-investigation" id="toc-statistical-investigation" class="nav-link" data-scroll-target="#statistical-investigation">Statistical Investigation</a>
  <ul class="collapse">
  <li><a href="#fischers-exact-test" id="toc-fischers-exact-test" class="nav-link" data-scroll-target="#fischers-exact-test">Fischer’s Exact Test</a></li>
  <li><a href="#the-effect-of-the-distance-between-bubbles-and-motifs" id="toc-the-effect-of-the-distance-between-bubbles-and-motifs" class="nav-link" data-scroll-target="#the-effect-of-the-distance-between-bubbles-and-motifs">The effect of the distance between bubbles and motifs</a></li>
  </ul></li>
  <li><a href="#linear-regressions" id="toc-linear-regressions" class="nav-link" data-scroll-target="#linear-regressions">Linear Regressions</a></li>
  <li><a href="#correlations-of-motif-positions-and-flipping-features" id="toc-correlations-of-motif-positions-and-flipping-features" class="nav-link" data-scroll-target="#correlations-of-motif-positions-and-flipping-features">Correlations of Motif positions and Flipping features</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Analysis of gcPBM data</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<style>
  code {
    white-space : pre-wrap !important;
    word-break: break-word;
  }
</style>
<section id="overview-on-the-dataset-and-data-processing" class="level2">
<h2 class="anchored" data-anchor-id="overview-on-the-dataset-and-data-processing">Overview on the Dataset and Data Processing</h2>
<section id="bubbles" class="level3">
<h3 class="anchored" data-anchor-id="bubbles">Bubbles</h3>
<p>We acquired gcPBM datasets for three TFs (MAX, MAD, MYC) from Alexandrov Boian’s lab. The data was about yay high and yay wide. We also acquired corresponding feature data for flipping and bubbles.</p>
<p>The bubble data, when acquired, was in the form of matrices with 130 rows and 36 columns. The individual columns represented base-pair positions, and rows represented time in picoseconds. The values in the matrix represented the distance between complementary base pairs at that position and time.</p>
<div id="fig-Bubble_Graphs" class="quarto-layout-panel">
<figure class="figure">
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-figure quarto-figure-center" style="flex-basis: 33.3%;justify-content: center;">
<figure class="figure">
<p><a href="figs/phase3/bubble_raw/image.png" class="lightbox" data-gallery="raw_bubble"><img src="figs/phase3/bubble_raw/image.png" class="img-fluid figure-img"></a></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center" style="flex-basis: 33.3%;justify-content: center;">
<figure class="figure">
<p><a href="figs/phase3/bubble_raw/image1.png" class="lightbox" data-gallery="raw_bubble"><img src="figs/phase3/bubble_raw/image1.png" class="img-fluid figure-img"></a></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center" style="flex-basis: 33.3%;justify-content: center;">
<figure class="figure">
<p><a href="figs/phase3/bubble_raw/image2.png" class="lightbox" data-gallery="raw_bubble"><img src="figs/phase3/bubble_raw/image2.png" class="img-fluid figure-img"></a></p>
</figure>
</div>
</div>
<p></p><figcaption class="figure-caption">Figure&nbsp;1: Bubble heatmaps of sequences with the highest bubble values across all TFs (TF_partition_id - bubble_value)</figcaption><p></p>
</figure>
</div>
<p>We investigated multiple methods to optimally encode the bubble data to provide it to the CNN as an additional input channel alongside one-hot encoded sequence data.</p>
<div id="fig-Affinity_and_Bubble_Histograms" class="quarto-layout-panel">
<figure class="figure">
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-figure quarto-figure-center" style="flex-basis: 48.0%;justify-content: center;">
<figure class="figure">
<p><a href="figs/phase3/affinity_histogram.png" class="lightbox" data-gallery="histogram" title="Normalized affinity values for each sequence in each TF"><img src="figs/phase3/affinity_histogram.png" class="img-fluid figure-img"></a></p>
<figcaption class="figure-caption">Normalized affinity values for each sequence in each TF</figcaption>
</figure>
</div>
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 4.0%;justify-content: center;">
<p>&nbsp;</p>
</div>
<div class="quarto-figure quarto-figure-center" style="flex-basis: 48.0%;justify-content: center;">
<figure class="figure">
<p><a href="figs/phase3/bubble_histogram.png" class="lightbox" data-gallery="histogram" title="Logarithmic distribution of all bubble values for each TF"><img src="figs/phase3/bubble_histogram.png" class="img-fluid figure-img"></a></p>
<figcaption class="figure-caption">Logarithmic distribution of all bubble values for each TF</figcaption>
</figure>
</div>
</div>
<p></p><figcaption class="figure-caption">Figure&nbsp;2: gcPBM exploration of affinity and bubble data values</figcaption><p></p>
</figure>
</div>
<p>We experimented with taking the average value across all time points, filtering this mean curve, and using the maxima of the resultant curve to delineate a bubble across a certain base pair window focused on the maxima.</p>
<div id="fig-Processed_Bubble" class="quarto-layout-panel">
<figure class="figure">
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-figure quarto-figure-center" style="flex-basis: 48.0%;justify-content: center;">
<figure class="figure">
<p><a href="figs/phase3/processed_bubble.png" class="lightbox" data-gallery="processed_bubble"><img src="figs/phase3/processed_bubble.png" class="img-fluid figure-img"></a></p>
</figure>
</div>
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 4.0%;justify-content: center;">
<p>&nbsp;</p>
</div>
<div class="quarto-figure quarto-figure-center" style="flex-basis: 48.0%;justify-content: center;">
<figure class="figure">
<p><a href="figs/phase3/processed_bubble 2.png" class="lightbox" data-gallery="processed_bubble"><img src="figs/phase3/processed_bubble 2.png" class="img-fluid figure-img"></a></p>
</figure>
</div>
</div>
<p></p><figcaption class="figure-caption">Figure&nbsp;3: Initial attempt at processing bubble data (left to right: heatmap, mean curve, filtered curve, maxima of filtered curve with 3 base pair window)</figcaption><p></p>
</figure>
</div>
<p>Using the different variants in our gridsearch for best model parameters revealed no clear winner. We decided to investigate the bubble percentile values and eventually decided on all locations at timepoint 0 (t=0) with a value &gt; 0.03 as bubbles.</p>
</section>
<section id="motifs" class="level3">
<h3 class="anchored" data-anchor-id="motifs">Motifs</h3>
<p>We tried to acquire motifs for the TFs from the JASPAR database, but we were unable to find motifs for MAD. We then used the motifs for MAX and MYC to scan for motif presence in the sequences of the gcPBM data. We used the FIMO tool from the MEME suite to scan for motif presence. We used the following parameters:</p>
<p>TBD</p>
</section>
<section id="processed-data" class="level3">
<h3 class="anchored" data-anchor-id="processed-data">Processed Data</h3>
<p>Once we had the motifs and bubbles identified and tied to base pair locations, we did some basic exploration of the sequences and how motifs and bubbles presented themselves.</p>
<p>Top 5 sequences per TF in terms of binding affinity that also have bubble features:</p>
<p>TF = max</p>
<pre class="{code}"><code>1.  seq_id = 4831
1.  DNA_sequence    = TGTGGCTTATCAAACCACGTGGTTTATCAAAATAAT
1.  Bubble_sequence = ------+++++-----------++++++++++++--
1.  Motif_sequence  = -------------++++++++++-------------
2.  seq_id = 4187
2.  DNA_sequence    = GAATTTGACCAAAACCACGTGGTTTGGCTGCAAGTT
2.  Bubble_sequence = -+++--------------------------------
2.  Motif_sequence  = -------------++++++++++-------------
3.  seq_id = 5934
3.  DNA_sequence    = ATCAGGAATCTGGACCACGTGGTCAGGCAGAACCAG
3.  Bubble_sequence = ------+-----------------------------
3.  Motif_sequence  = -------------++++++++++-------------
4.  seq_id = 4471
4.  DNA_sequence    = GGGCTTATGAGAGACCACGTGGTCTTGTGCTTTAGT
4.  Bubble_sequence = ------+-----------------------------
4.  Motif_sequence  = -------------++++++++++-------------
5.  seq_id = 8206
5.  DNA_sequence    = ACAACCACAGAAAACCACGTGGTTTGCTCTGAAATT
5.  Bubble_sequence = ----------++-------------------++---
5.  Motif_sequence  = -------------++++++++++-------------</code></pre>
<p>TF = myc</p>
<pre class="{code}"><code>1.  seq_id = 5288
1.  DNA_sequence    = TGTGGCTTATCAAACCACGTGGTTTATCAAAATAAT
1.  Bubble_sequence = ------++++++----------++++++++++++--
1.  Motif_sequence  = ------------++++++++++++------------
2.  seq_id = 3034
2.  DNA_sequence    = GAATTTGACCAAAACCACGTGGTTTGGCTGCAAGTT
2.  Bubble_sequence = --++--------------------------------
2.  Motif_sequence  = ------------++++++++++++------------
3.  seq_id = 3153
3.  DNA_sequence    = CTAGAACCTTTGTACCACGTGGCATTTTTTTAAACT
3.  Bubble_sequence = -----------------------+++++++++----
3.  Motif_sequence  = ------------++++++++++++------------
4.  seq_id = 909
4.  DNA_sequence    = GGGCTTATGAGAGACCACGTGGTCTTGTGCTTTAGT
4.  Bubble_sequence = -----++-----------------------------
4.  Motif_sequence  = ------------++++++++++++------------
5.  seq_id = 3063
5.  DNA_sequence    = ATCAGGAATCTGGACCACGTGGTCAGGCAGAACCAG
5.  Bubble_sequence = ------+-----------------------------
5.  Motif_sequence  = ------------++++++++++++------------</code></pre>
</section>
</section>
<section id="machine-learning-investigation" class="level2">
<h2 class="anchored" data-anchor-id="machine-learning-investigation">Machine Learning Investigation</h2>
<p>We used a CNN to predict the binding affinity of a TF to a DNA sequence. Our initial models were trained using only the sequences, using a gridsearch to find the best model parameters. We used the following parameters:</p>
<pre class="{code}"><code>kernel size: 3, 5, 7, 9
number of filters: 16, 32, 64, 128
number of layers: 1, 2, 3
pooling: 3, 1</code></pre>
<p>Across all TFs, the best model parameters were a kernel size of 7, 64 filters in the first layer, 3 layers, and no pooling (1). We then used these parameters to train models with the bubble data as an additional input channel, and we contrasted the performance of these models.</p>
<div id="fig-Model_Scatterplots" class="quarto-layout-panel">
<figure class="figure">
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-figure quarto-figure-center" style="flex-basis: 31.6%;justify-content: center;">
<figure class="figure">
<p><a href="figs/phase3/scatterplots_7/mad_seq_only_1_7_64_3.png" class="lightbox" data-gallery="scatterplot"><img src="figs/phase3/scatterplots_7/mad_seq_only_1_7_64_3.png" class="img-fluid figure-img"></a></p>
</figure>
</div>
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 2.6%;justify-content: center;">
<p>&nbsp;</p>
</div>
<div class="quarto-figure quarto-figure-center" style="flex-basis: 31.6%;justify-content: center;">
<figure class="figure">
<p><a href="figs/phase3/scatterplots_7/max_seq_only_1_7_64_3.png" class="lightbox" data-gallery="scatterplot"><img src="figs/phase3/scatterplots_7/max_seq_only_1_7_64_3.png" class="img-fluid figure-img"></a></p>
</figure>
</div>
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 2.6%;justify-content: center;">
<p>&nbsp;</p>
</div>
<div class="quarto-figure quarto-figure-center" style="flex-basis: 31.6%;justify-content: center;">
<figure class="figure">
<p><a href="figs/phase3/scatterplots_7/myc_seq_only_1_7_64_3.png" class="lightbox" data-gallery="scatterplot"><img src="figs/phase3/scatterplots_7/myc_seq_only_1_7_64_3.png" class="img-fluid figure-img"></a></p>
</figure>
</div>
</div>
<p></p><figcaption class="figure-caption">Figure&nbsp;4: Scatterplot showcasing the parameters and results from the best model for each TF</figcaption><p></p>
</figure>
</div>
<p>The unanimous conclusion was that the models trained with the bubble data performed worse than the models trained with only the sequence data. We then revised the way we were approaching the bubble data.</p>
<p>After observing the distribution patterns for motifs and bubbles, we decided to redo the gridsearch by searching in the range of kernel size = 6, 8, 10, and comparing sequence only models to models including bubble features.</p>
<p>TF: Mad</p>
<pre class="{code}"><code>Best to worst model
Model: seq_only_1_6_64_3, MSE: 0.025164015591144562, r2: 0.9751617209427638
Model: seq_only_1_8_64_3, MSE: 0.02559092827141285, r2: 0.9747403303054708
Model: bubble_linear_1_6_64_3, MSE: 0.025975389406085014, r2: 0.9743608481893277
Model: seq_only_1_10_64_3, MSE: 0.025992512702941895, r2: 0.9743439469430584
Model: bubble_linear_1_8_64_3, MSE: 0.02697504125535488, r2: 0.9733741378803168
Model: bubble_linear_1_10_64_3, MSE: 0.02804403379559517, r2: 0.9723189799559782</code></pre>
<div id="fig-Model_Scatterplots" class="quarto-layout-panel">
<figure class="figure">
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-figure quarto-figure-center" style="flex-basis: 33.3%;justify-content: center;">
<figure class="figure">
<p><a href="figs/phase3/scatterplots_6810/mad_seq_only_1_6_64_3.png" class="lightbox" data-gallery="scatterplot1"><img src="figs/phase3/scatterplots_6810/mad_seq_only_1_6_64_3.png" class="img-fluid figure-img"></a></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center" style="flex-basis: 33.3%;justify-content: center;">
<figure class="figure">
<p><a href="figs/phase3/scatterplots_6810/mad_seq_only_1_8_64_3.png" class="lightbox" data-gallery="scatterplot1"><img src="figs/phase3/scatterplots_6810/mad_seq_only_1_8_64_3.png" class="img-fluid figure-img"></a></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center" style="flex-basis: 33.3%;justify-content: center;">
<figure class="figure">
<p><a href="figs/phase3/scatterplots_6810/mad_seq_only_1_10_64_3.png" class="lightbox" data-gallery="scatterplot1"><img src="figs/phase3/scatterplots_6810/mad_seq_only_1_10_64_3.png" class="img-fluid figure-img"></a></p>
</figure>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-figure quarto-figure-center" style="flex-basis: 33.3%;justify-content: center;">
<figure class="figure">
<p><a href="figs/phase3/scatterplots_6810/mad_bubble_linear_1_6_64_3.png" class="lightbox" data-gallery="scatterplot1"><img src="figs/phase3/scatterplots_6810/mad_bubble_linear_1_6_64_3.png" class="img-fluid figure-img"></a></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center" style="flex-basis: 33.3%;justify-content: center;">
<figure class="figure">
<p><a href="figs/phase3/scatterplots_6810/mad_bubble_linear_1_8_64_3.png" class="lightbox" data-gallery="scatterplot1"><img src="figs/phase3/scatterplots_6810/mad_bubble_linear_1_8_64_3.png" class="img-fluid figure-img"></a></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center" style="flex-basis: 33.3%;justify-content: center;">
<figure class="figure">
<p><a href="figs/phase3/scatterplots_6810/mad_bubble_linear_1_10_64_3.png" class="lightbox" data-gallery="scatterplot1"><img src="figs/phase3/scatterplots_6810/mad_bubble_linear_1_10_64_3.png" class="img-fluid figure-img"></a></p>
</figure>
</div>
</div>
<p></p><figcaption class="figure-caption">Figure&nbsp;5: Model comparison using Scatterplots across MAD</figcaption><p></p>
</figure>
</div>
<p>TF: Max</p>
<pre class="{code}"><code>Best to worst model
Model: bubble_linear_1_6_64_3, MSE: 0.04008778929710388, r2: 0.960649553399618
Model: seq_only_1_6_64_3, MSE: 0.04050329700112343, r2: 0.9602416902661023
Model: seq_only_1_8_64_3, MSE: 0.040858376771211624, r2: 0.9598931401387409
Model: seq_only_1_10_64_3, MSE: 0.04170531406998634, r2: 0.9590617832927455
Model: bubble_linear_1_8_64_3, MSE: 0.0423930287361145, r2: 0.9583867144681565
Model: bubble_linear_1_10_64_3, MSE: 0.043693408370018005, r2: 0.9571102520169519</code></pre>
<div id="fig-Model_Scatterplots" class="quarto-layout-panel">
<figure class="figure">
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-figure quarto-figure-center" style="flex-basis: 33.3%;justify-content: center;">
<figure class="figure">
<p><a href="figs/phase3/scatterplots_6810/max_seq_only_1_6_64_3.png" class="lightbox" data-gallery="scatterplot2"><img src="figs/phase3/scatterplots_6810/max_seq_only_1_6_64_3.png" class="img-fluid figure-img"></a></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center" style="flex-basis: 33.3%;justify-content: center;">
<figure class="figure">
<p><a href="figs/phase3/scatterplots_6810/max_seq_only_1_8_64_3.png" class="lightbox" data-gallery="scatterplot2"><img src="figs/phase3/scatterplots_6810/max_seq_only_1_8_64_3.png" class="img-fluid figure-img"></a></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center" style="flex-basis: 33.3%;justify-content: center;">
<figure class="figure">
<p><a href="figs/phase3/scatterplots_6810/max_seq_only_1_10_64_3.png" class="lightbox" data-gallery="scatterplot2"><img src="figs/phase3/scatterplots_6810/max_seq_only_1_10_64_3.png" class="img-fluid figure-img"></a></p>
</figure>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-figure quarto-figure-center" style="flex-basis: 33.3%;justify-content: center;">
<figure class="figure">
<p><a href="figs/phase3/scatterplots_6810/max_bubble_linear_1_6_64_3.png" class="lightbox" data-gallery="scatterplot2"><img src="figs/phase3/scatterplots_6810/max_bubble_linear_1_6_64_3.png" class="img-fluid figure-img"></a></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center" style="flex-basis: 33.3%;justify-content: center;">
<figure class="figure">
<p><a href="figs/phase3/scatterplots_6810/max_bubble_linear_1_8_64_3.png" class="lightbox" data-gallery="scatterplot2"><img src="figs/phase3/scatterplots_6810/max_bubble_linear_1_8_64_3.png" class="img-fluid figure-img"></a></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center" style="flex-basis: 33.3%;justify-content: center;">
<figure class="figure">
<p><a href="figs/phase3/scatterplots_6810/max_bubble_linear_1_10_64_3.png" class="lightbox" data-gallery="scatterplot2"><img src="figs/phase3/scatterplots_6810/max_bubble_linear_1_10_64_3.png" class="img-fluid figure-img"></a></p>
</figure>
</div>
</div>
<p></p><figcaption class="figure-caption">Figure&nbsp;6: Model comparison using Scatterplots across MAX</figcaption><p></p>
</figure>
</div>
<p>TF: Myc</p>
<pre class="{code}"><code>Best to worst model
Model: seq_only_1_8_64_3, MSE: 0.05695045739412308, r2: 0.9444290673365138
Model: seq_only_1_10_64_3, MSE: 0.05770198628306389, r2: 0.9436957503397924
Model: bubble_linear_1_8_64_3, MSE: 0.05852825567126274, r2: 0.9428894908346188
Model: seq_only_1_6_64_3, MSE: 0.060217346996068954, r2: 0.9412413163392869
Model: bubble_linear_1_6_64_3, MSE: 0.060966987162828445, r2: 0.940509840597948
Model: bubble_linear_1_10_64_3, MSE: 0.06115123629570007, r2: 0.9403300450380424</code></pre>
<div id="fig-Model_Scatterplots" class="quarto-layout-panel">
<figure class="figure">
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-figure quarto-figure-center" style="flex-basis: 33.3%;justify-content: center;">
<figure class="figure">
<p><a href="figs/phase3/scatterplots_6810/myc_seq_only_1_6_64_3.png" class="lightbox" data-gallery="scatterplot3"><img src="figs/phase3/scatterplots_6810/myc_seq_only_1_6_64_3.png" class="img-fluid figure-img"></a></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center" style="flex-basis: 33.3%;justify-content: center;">
<figure class="figure">
<p><a href="figs/phase3/scatterplots_6810/myc_seq_only_1_8_64_3.png" class="lightbox" data-gallery="scatterplot3"><img src="figs/phase3/scatterplots_6810/myc_seq_only_1_8_64_3.png" class="img-fluid figure-img"></a></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center" style="flex-basis: 33.3%;justify-content: center;">
<figure class="figure">
<p><a href="figs/phase3/scatterplots_6810/myc_seq_only_1_10_64_3.png" class="lightbox" data-gallery="scatterplot3"><img src="figs/phase3/scatterplots_6810/myc_seq_only_1_10_64_3.png" class="img-fluid figure-img"></a></p>
</figure>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-figure quarto-figure-center" style="flex-basis: 33.3%;justify-content: center;">
<figure class="figure">
<p><a href="figs/phase3/scatterplots_6810/myc_bubble_linear_1_6_64_3.png" class="lightbox" data-gallery="scatterplot3"><img src="figs/phase3/scatterplots_6810/myc_bubble_linear_1_6_64_3.png" class="img-fluid figure-img"></a></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center" style="flex-basis: 33.3%;justify-content: center;">
<figure class="figure">
<p><a href="figs/phase3/scatterplots_6810/myc_bubble_linear_1_8_64_3.png" class="lightbox" data-gallery="scatterplot3"><img src="figs/phase3/scatterplots_6810/myc_bubble_linear_1_8_64_3.png" class="img-fluid figure-img"></a></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center" style="flex-basis: 33.3%;justify-content: center;">
<figure class="figure">
<p><a href="figs/phase3/scatterplots_6810/myc_bubble_linear_1_10_64_3.png" class="lightbox" data-gallery="scatterplot3"><img src="figs/phase3/scatterplots_6810/myc_bubble_linear_1_10_64_3.png" class="img-fluid figure-img"></a></p>
</figure>
</div>
</div>
<p></p><figcaption class="figure-caption">Figure&nbsp;7: Model comparison using Scatterplots across MYC</figcaption><p></p>
</figure>
</div>
<p>At a kernel size of 6, MAX seems to benefit from the inclusion of bubble data, while MAD and MYC do not seem to benefit from bubble data at all. The results were not very conclusive, so we decided to focus on a more statistical approach.</p>
</section>
<section id="statistical-investigation" class="level2">
<h2 class="anchored" data-anchor-id="statistical-investigation">Statistical Investigation</h2>
<p>We conducted statistical investigations into the relationship between bubbles, motifs, and binding affinity. Our first investigation was a t-test to see if the presence of a bubble significantly affected binding affinity.</p>
<pre class="{code}"><code>MAD
Ttest_indResult(statistic=8.007428268390886, pvalue=1.3472019719992042e-15)

MAX
Ttest_indResult(statistic=12.270239455302246, pvalue=2.5385730406239527e-34)

MYC
Ttest_indResult(statistic=5.339201658963574, pvalue=9.632386462328061e-08)</code></pre>
<div id="fig-box_and_histograms" class="quarto-layout-panel">
<figure class="figure">
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-figure quarto-figure-center" style="flex-basis: 50.0%;justify-content: center;">
<figure class="figure">
<p><a href="figs/phase3/bubble_threshold_ttest/mad_box.png" class="lightbox" data-gallery="boxplot_histogram"><img src="figs/phase3/bubble_threshold_ttest/mad_box.png" class="img-fluid figure-img"></a></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center" style="flex-basis: 50.0%;justify-content: center;">
<figure class="figure">
<p><a href="figs/phase3/bubble_threshold_ttest/mad_hist.png" class="lightbox" data-gallery="boxplot_histogram"><img src="figs/phase3/bubble_threshold_ttest/mad_hist.png" class="img-fluid figure-img"></a></p>
</figure>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-figure quarto-figure-center" style="flex-basis: 50.0%;justify-content: center;">
<figure class="figure">
<p><a href="figs/phase3/bubble_threshold_ttest/max_box.png" class="lightbox" data-gallery="boxplot_histogram"><img src="figs/phase3/bubble_threshold_ttest/max_box.png" class="img-fluid figure-img"></a></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center" style="flex-basis: 50.0%;justify-content: center;">
<figure class="figure">
<p><a href="figs/phase3/bubble_threshold_ttest/max_hist.png" class="lightbox" data-gallery="boxplot_histogram"><img src="figs/phase3/bubble_threshold_ttest/max_hist.png" class="img-fluid figure-img"></a></p>
</figure>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-figure quarto-figure-center" style="flex-basis: 50.0%;justify-content: center;">
<figure class="figure">
<p><a href="figs/phase3/bubble_threshold_ttest/myc_box.png" class="lightbox" data-gallery="boxplot_histogram"><img src="figs/phase3/bubble_threshold_ttest/myc_box.png" class="img-fluid figure-img"></a></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center" style="flex-basis: 50.0%;justify-content: center;">
<figure class="figure">
<p><a href="figs/phase3/bubble_threshold_ttest/myc_hist.png" class="lightbox" data-gallery="boxplot_histogram"><img src="figs/phase3/bubble_threshold_ttest/myc_hist.png" class="img-fluid figure-img"></a></p>
</figure>
</div>
</div>
<p></p><figcaption class="figure-caption">Figure&nbsp;8: Boxplots and histograms of binding affinity for sequences with and without bubbles.</figcaption><p></p>
</figure>
</div>
<p>There appeared to be a significant relationship between bubble presence and binding affinity.</p>
<p>Since we were unable to find a motif for MAD from JASPAR, the statistical analyses investigating motifs were conducted with the data for MAX and MYC only.</p>
<section id="fischers-exact-test" class="level3">
<h3 class="anchored" data-anchor-id="fischers-exact-test">Fischer’s Exact Test</h3>
<p>We conducted a Fischer’s Exact Test to see if the presence of a bubble and motif together significantly affected binding affinity. We counted the base pair positions across our entire dataset that had a bubble, motif, or both, and used these counts to construct a contingency table for the Fischer’s Exact Test.</p>
<div id="fig-Fischer" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<pre class="{code}"><code>{'max': {'L': 308448, 'B': 12247, 'M': 20190, 'x': 140},
 'myc': {'L': 249336, 'B': 10106, 'M': 27984, 'x': 194}}</code></pre>
<pre class="{code}"><code>TF: max
Contingency Table:
         | Bubble+ | Bubble-
Motif+   |   140   |   20050
Motif-   |   12107   |   276151
Odds Ratio: 0.15926624341104784
P-value: 2.0930098404665845e-196

TF: myc
Contingency Table:
         | Bubble+ | Bubble-
Motif+   |   194   |   27790
Motif-   |   9912   |   211440
Odds Ratio: 0.14891520370262265
P-value: 4.150518659943197e-290</code></pre>
<figcaption class="figure-caption">Figure&nbsp;9: Fisher’s Exact test using all sequences: L = total length of sequences, B = coordinates with a bubble, M = coordinates with a motif, x = coordinates with both a motif and bubble.</figcaption>
</figure>
</div>
<p>The results of the Fischer’s Exact Test showed that the presence of a bubble and motif together significantly affected binding affinity.</p>
</section>
<section id="the-effect-of-the-distance-between-bubbles-and-motifs" class="level3">
<h3 class="anchored" data-anchor-id="the-effect-of-the-distance-between-bubbles-and-motifs">The effect of the distance between bubbles and motifs</h3>
<p>We then investigated the effect of the distance between bubbles and motifs on binding affinity. To this end, we categorized the top 25% of sequences by binding affinity to be positive sequences. We then defined close distances between motifs and bubbles using a maximum threshold, and then performed a fischer’s exact test at different thresholds.</p>
<div id="fig-Fischer" class="quarto-layout-panel">
<figure class="figure">
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-figure quarto-figure-center" style="flex-basis: 48.0%;justify-content: center;">
<figure class="figure">
<p><a href="figs/phase3/binding_posvneg_distance/max_posvneg.png" class="lightbox" data-gallery="bubble_motif_distance"><img src="figs/phase3/binding_posvneg_distance/max_posvneg.png" class="img-fluid figure-img"></a></p>
</figure>
</div>
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 4.0%;justify-content: center;">
<p>&nbsp;</p>
</div>
<div class="quarto-figure quarto-figure-center" style="flex-basis: 48.0%;justify-content: center;">
<figure class="figure">
<p><a href="figs/phase3/binding_posvneg_distance/myc_posvneg.png" class="lightbox" data-gallery="bubble_motif_distance"><img src="figs/phase3/binding_posvneg_distance/myc_posvneg.png" class="img-fluid figure-img"></a></p>
</figure>
</div>
</div>
<p></p><figcaption class="figure-caption">Figure&nbsp;10: Histograms showing the distribution of distances between bubbles and motifs for positive and negative sequences (top 25% of sequences by binding affinity = positive).</figcaption><p></p>
</figure>
</div>
<div id="tbl-letters" class="anchored">
<table class="table">
<caption>Table&nbsp;1: Summary of the results of the Fischer’s Exact Test at different distance thresholds.</caption>
<thead>
<tr class="header">
<th>Threshold</th>
<th>Max Odds Ratio (P-value)</th>
<th>Myc Odds Ratio (P-value)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>≤ 1</td>
<td>inf (0.334863)</td>
<td>1.839879 (0.000567)</td>
</tr>
<tr class="even">
<td>≤ 2</td>
<td>inf (0.162437)</td>
<td>1.396816 (0.035636)</td>
</tr>
<tr class="odd">
<td>≤ 3</td>
<td>inf (0.063436)</td>
<td>0.991813 (1.000000)</td>
</tr>
<tr class="even">
<td>≤ 4</td>
<td>1.939929 (0.658042)</td>
<td>0.888585 (0.474167)</td>
</tr>
<tr class="odd">
<td>≤ 5</td>
<td>1.413462 (0.658332)</td>
<td>0.834876 (0.306648)</td>
</tr>
</tbody>
</table>
</div>
<p>The results were significant for MYC at the 1 and 2 thresholds at the 0.05 significance level. The lack of negative values were thought to contribute to the lack of significance for MAX, so the threshold for positive sequences was lowered to the top 10% of sequences and the analysis was repeated.</p>
<div id="fig-Fischer" class="quarto-layout-panel">
<figure class="figure">
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-figure quarto-figure-center" style="flex-basis: 48.0%;justify-content: center;">
<figure class="figure">
<p><a href="figs/phase3/binding_posvneg_distance/max_posvneg_10.png" class="lightbox" data-gallery="bubble_motif_distance2"><img src="figs/phase3/binding_posvneg_distance/max_posvneg_10.png" class="img-fluid figure-img"></a></p>
</figure>
</div>
<div class="quarto-figure-spacer quarto-layout-cell" style="flex-basis: 4.0%;justify-content: center;">
<p>&nbsp;</p>
</div>
<div class="quarto-figure quarto-figure-center" style="flex-basis: 48.0%;justify-content: center;">
<figure class="figure">
<p><a href="figs/phase3/binding_posvneg_distance/myc_posvneg_10.png" class="lightbox" data-gallery="bubble_motif_distance2"><img src="figs/phase3/binding_posvneg_distance/myc_posvneg_10.png" class="img-fluid figure-img"></a></p>
</figure>
</div>
</div>
<p></p><figcaption class="figure-caption">Figure&nbsp;11: Histograms showing the distribution of distances between bubbles and motifs for positive and negative sequences (top 10% of sequences by binding affinity = positive).</figcaption><p></p>
</figure>
</div>
<div id="tbl-letters" class="anchored">
<table class="table">
<caption>Table&nbsp;2: Summary of the results of the Fischer’s Exact Test at different distance thresholds.</caption>
<thead>
<tr class="header">
<th>Threshold</th>
<th>Max Odds Ratio (P-value)</th>
<th>Myc Odds Ratio (P-value)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>≤ 1</td>
<td>1.335677 (0.108784)</td>
<td>1.184628 (0.341806)</td>
</tr>
<tr class="even">
<td>≤ 2</td>
<td>1.375877 (0.053442)</td>
<td>0.970310 (0.872669)</td>
</tr>
<tr class="odd">
<td>≤ 3</td>
<td>1.139963 (0.433580)</td>
<td>0.779458 (0.133276)</td>
</tr>
<tr class="even">
<td>≤ 4</td>
<td>1.034811 (0.874747)</td>
<td>0.680505 (0.021350)</td>
</tr>
<tr class="odd">
<td>≤ 5</td>
<td>0.870851 (0.450992)</td>
<td>0.624400 (0.006396)</td>
</tr>
</tbody>
</table>
</div>
<p>The results were significant for MYC at the 4 and 5 threshold at the 0.05 significance level. Results for MAX are still not significant.</p>
</section>
</section>
<section id="linear-regressions" class="level2">
<h2 class="anchored" data-anchor-id="linear-regressions">Linear Regressions</h2>
<p>We then conducted linear regressions to see the degree to which the presence of a bubble, motif, and an interaction feature significantly affected binding affinity.</p>
<div id="tbl-panel" class="tbl-parent quarto-layout-panel anchored">
<div class="quarto-layout-row quarto-layout-valign-top">
<div id="tbl-panel1" class="quarto-layout-cell quarto-layout-cell-subref anchored" data-ref-parent="tbl-panel" style="flex-basis: 100.0%;justify-content: center;">
<table class="table">
<caption>(a) Linear regression results for MAX (R-squared = 0.708, F-statistic = 6910).</caption>
<colgroup>
<col style="width: 17%">
<col style="width: 19%">
<col style="width: 18%">
<col style="width: 17%">
<col style="width: 18%">
<col style="width: 4%">
<col style="width: 4%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: left;">coef</th>
<th style="text-align: left;">std err</th>
<th style="text-align: left;">t</th>
<th style="text-align: left;">P&gt;</th>
<th style="text-align: left;">t</th>
<th style="text-align: left;"></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">const</td>
<td style="text-align: left;">-0.5620</td>
<td style="text-align: left;">0.008</td>
<td style="text-align: left;">-68.981</td>
<td style="text-align: left;">0.000</td>
<td style="text-align: left;">-0.578</td>
<td style="text-align: left;">-0.546</td>
</tr>
<tr class="even">
<td style="text-align: left;">motif_binary</td>
<td style="text-align: left;">2.0056</td>
<td style="text-align: left;">0.017</td>
<td style="text-align: left;">119.550</td>
<td style="text-align: left;">0.000</td>
<td style="text-align: left;">1.973</td>
<td style="text-align: left;">2.039</td>
</tr>
<tr class="odd">
<td style="text-align: left;">bubble_binary</td>
<td style="text-align: left;">0.3153</td>
<td style="text-align: left;">0.014</td>
<td style="text-align: left;">22.047</td>
<td style="text-align: left;">0.000</td>
<td style="text-align: left;">0.287</td>
<td style="text-align: left;">0.343</td>
</tr>
<tr class="even">
<td style="text-align: left;">motif_and_bubble_binary</td>
<td style="text-align: left;">-0.1396</td>
<td style="text-align: left;">0.029</td>
<td style="text-align: left;">-4.737</td>
<td style="text-align: left;">0.000</td>
<td style="text-align: left;">-0.197</td>
<td style="text-align: left;">-0.082</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div id="tbl-panel2" class="quarto-layout-cell quarto-layout-cell-subref anchored" data-ref-parent="tbl-panel" style="flex-basis: 100.0%;justify-content: center;">
<table class="table">
<caption>(b) Linear regression results for MYC (R-squared = 0.406, F-statistic = 1579).</caption>
<colgroup>
<col style="width: 17%">
<col style="width: 19%">
<col style="width: 18%">
<col style="width: 17%">
<col style="width: 18%">
<col style="width: 4%">
<col style="width: 4%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: left;">coef</th>
<th style="text-align: left;">std err</th>
<th style="text-align: left;">t</th>
<th style="text-align: left;">P&gt;</th>
<th style="text-align: left;">t</th>
<th style="text-align: left;"></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">const</td>
<td style="text-align: left;">-0.5018</td>
<td style="text-align: left;">0.014</td>
<td style="text-align: left;">-35.384</td>
<td style="text-align: left;">0.000</td>
<td style="text-align: left;">-0.530</td>
<td style="text-align: left;">-0.474</td>
</tr>
<tr class="even">
<td style="text-align: left;">motif_binary</td>
<td style="text-align: left;">1.3369</td>
<td style="text-align: left;">0.024</td>
<td style="text-align: left;">55.488</td>
<td style="text-align: left;">0.000</td>
<td style="text-align: left;">1.290</td>
<td style="text-align: left;">1.384</td>
</tr>
<tr class="odd">
<td style="text-align: left;">bubble_binary</td>
<td style="text-align: left;">0.1645</td>
<td style="text-align: left;">0.024</td>
<td style="text-align: left;">6.899</td>
<td style="text-align: left;">0.000</td>
<td style="text-align: left;">0.118</td>
<td style="text-align: left;">0.211</td>
</tr>
<tr class="even">
<td style="text-align: left;">motif_and_bubble_binary</td>
<td style="text-align: left;">0.0270</td>
<td style="text-align: left;">0.042</td>
<td style="text-align: left;">0.648</td>
<td style="text-align: left;">0.517</td>
<td style="text-align: left;">-0.055</td>
<td style="text-align: left;">0.109</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="panel-caption table-caption">
<p>Table&nbsp;3: Linear Regression of all sequences with Motif, Bubble, and Interaction Features.</p>
</div>
</div>
<div id="tbl-panel" class="tbl-parent quarto-layout-panel anchored">
<div class="quarto-layout-row quarto-layout-valign-top">
<div id="tbl-panel1" class="quarto-layout-cell quarto-layout-cell-subref anchored" data-ref-parent="tbl-panel" style="flex-basis: 100.0%;justify-content: center;">
<table class="table">
<caption>(a) Linear regression results for MAX (R-squared = 0.017, F-statistic = 150.6).</caption>
<colgroup>
<col style="width: 22%">
<col style="width: 12%">
<col style="width: 12%">
<col style="width: 12%">
<col style="width: 12%">
<col style="width: 12%">
<col style="width: 12%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: left;">coef</th>
<th style="text-align: left;">std err</th>
<th style="text-align: left;">t</th>
<th style="text-align: left;">P&gt;</th>
<th style="text-align: left;">t</th>
<th style="text-align: left;">[0.025</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">const</td>
<td style="text-align: left;">-0.0890</td>
<td style="text-align: left;">0.013</td>
<td style="text-align: left;">-6.817</td>
<td style="text-align: left;">0.000</td>
<td style="text-align: left;">-0.115</td>
<td style="text-align: left;">-0.063</td>
</tr>
<tr class="even">
<td style="text-align: left;">bubble_binary</td>
<td style="text-align: left;">0.2812</td>
<td style="text-align: left;">0.023</td>
<td style="text-align: left;">12.270</td>
<td style="text-align: left;">0.000</td>
<td style="text-align: left;">0.236</td>
<td style="text-align: left;">0.326</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div id="tbl-panel2" class="quarto-layout-cell quarto-layout-cell-subref anchored" data-ref-parent="tbl-panel" style="flex-basis: 100.0%;justify-content: center;">
<table class="table">
<caption>(b) Linear regression results for MYC (R-squared = 0.004, F-statistic = 28.51).</caption>
<colgroup>
<col style="width: 23%">
<col style="width: 12%">
<col style="width: 12%">
<col style="width: 12%">
<col style="width: 12%">
<col style="width: 12%">
<col style="width: 12%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: left;">coef</th>
<th style="text-align: left;">std err</th>
<th style="text-align: left;">t</th>
<th style="text-align: left;">P&gt;</th>
<th style="text-align: left;">t</th>
<th style="text-align: left;">[0.025</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">const</td>
<td style="text-align: left;">-0.0386</td>
<td style="text-align: left;">0.015</td>
<td style="text-align: left;">-2.601</td>
<td style="text-align: left;">0.009</td>
<td style="text-align: left;">-0.068</td>
<td style="text-align: left;">-0.010</td>
</tr>
<tr class="even">
<td style="text-align: left;">bubble_binary</td>
<td style="text-align: left;">0.1351</td>
<td style="text-align: left;">0.025</td>
<td style="text-align: left;">5.339</td>
<td style="text-align: left;">0.000</td>
<td style="text-align: left;">0.086</td>
<td style="text-align: left;">0.185</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="panel-caption table-caption">
<p>Table&nbsp;4: Linear Regression of all sequences with Bubble Feature only.</p>
</div>
</div>
<p>The results of the linear regressions showed that the presence of a bubble and motif significantly affected binding affinity, and that the interaction feature did not significantly affect binding affinity. The regressions performed with bubble data showed the extent to which bubbles alone explain binding.</p>
</section>
<section id="correlations-of-motif-positions-and-flipping-features" class="level2">
<h2 class="anchored" data-anchor-id="correlations-of-motif-positions-and-flipping-features">Correlations of Motif positions and Flipping features</h2>
<p>Another hypothesis of ours was that flipping features inside or near motifs may determine binding affinity. To test this, we extracted all DNA sequences with motifs. Then we computed the average flipping feature across all positions in the motifs, and found the correlation between this average flipping feature and the binding affinity of the sequence for every sequence.</p>
<pre class="{code}"><code>TF = max
Pearson Correlation with Motif Flip Feature: Correlation = 0.21, p-value = 6.98e-21
Pearson Correlation with Motif Coord Feature: Correlation = 0.19, p-value = 1.32e-18
Pearson Correlation with Motif Coord^2 Feature: Correlation = 0.18, p-value = 1.02e-16</code></pre>
<pre class="{code}"><code>TF = myc
Pearson Correlation with Motif Flip Feature: Correlation = 0.10, p-value = 5.81e-07
Pearson Correlation with Motif Coord Feature: Correlation = 0.17, p-value = 4.37e-17
Pearson Correlation with Motif Coord^2 Feature: Correlation = 0.12, p-value = 6.81e-09</code></pre>
<div id="fig-Correlation" class="quarto-layout-panel">
<figure class="figure">
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-figure quarto-figure-center" style="flex-basis: 100.0%;justify-content: center;">
<figure class="figure">
<p><a href="figs/phase3/flip_correlation/max.png" class="lightbox" data-gallery="correlation" title="TF = max"><img src="figs/phase3/flip_correlation/max.png" class="img-fluid figure-img"></a></p>
<figcaption class="figure-caption">TF = max</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-figure quarto-figure-center" style="flex-basis: 100.0%;justify-content: center;">
<figure class="figure">
<p><a href="figs/phase3/flip_correlation/myc.png" class="lightbox" data-gallery="correlation" title="TF = myc"><img src="figs/phase3/flip_correlation/myc.png" class="img-fluid figure-img"></a></p>
<figcaption class="figure-caption">TF = myc</figcaption>
</figure>
</div>
</div>
<p></p><figcaption class="figure-caption">Figure&nbsp;12: Scatterplots showing the average flipping feature of motifs against binding affinity.</figcaption><p></p>
</figure>
</div>
<p>We then calculated correlations for each individual base pair position in the motif.</p>
<div id="fig-Correlation_bar" class="quarto-layout-panel">
<figure class="figure">
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-figure quarto-figure-center" style="flex-basis: 33.3%;justify-content: center;">
<figure class="figure">
<p><a href="figs/phase3/flip_correlation/flip_max.png" class="lightbox" data-gallery="correlation_bar"><img src="figs/phase3/flip_correlation/flip_max.png" class="img-fluid figure-img"></a></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center" style="flex-basis: 33.3%;justify-content: center;">
<figure class="figure">
<p><a href="figs/phase3/flip_correlation/coord_max.png" class="lightbox" data-gallery="correlation_bar"><img src="figs/phase3/flip_correlation/coord_max.png" class="img-fluid figure-img"></a></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center" style="flex-basis: 33.3%;justify-content: center;">
<figure class="figure">
<p><a href="figs/phase3/flip_correlation/coordsq_max.png" class="lightbox" data-gallery="correlation_bar"><img src="figs/phase3/flip_correlation/coordsq_max.png" class="img-fluid figure-img"></a></p>
</figure>
</div>
</div>
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="quarto-figure quarto-figure-center" style="flex-basis: 33.3%;justify-content: center;">
<figure class="figure">
<p><a href="figs/phase3/flip_correlation/flip_myc.png" class="lightbox" data-gallery="correlation_bar"><img src="figs/phase3/flip_correlation/flip_myc.png" class="img-fluid figure-img"></a></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center" style="flex-basis: 33.3%;justify-content: center;">
<figure class="figure">
<p><a href="figs/phase3/flip_correlation/coord_myc.png" class="lightbox" data-gallery="correlation_bar"><img src="figs/phase3/flip_correlation/coord_myc.png" class="img-fluid figure-img"></a></p>
</figure>
</div>
<div class="quarto-figure quarto-figure-center" style="flex-basis: 33.3%;justify-content: center;">
<figure class="figure">
<p><a href="figs/phase3/flip_correlation/coordsq_myc.png" class="lightbox" data-gallery="correlation_bar"><img src="figs/phase3/flip_correlation/coordsq_myc.png" class="img-fluid figure-img"></a></p>
</figure>
</div>
</div>
<p></p><figcaption class="figure-caption">Figure&nbsp;13: Barplots showing the correlation (and associated p-value) between flipping features and binding affinity for each base pair position in the motif.</figcaption><p></p>
</figure>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->
<script>var lightboxQuarto = GLightbox({"closeEffect":"zoom","openEffect":"zoom","descPosition":"bottom","loop":true,"selector":".lightbox"});</script>



</body></html>